python/cudf/cudf/core/multiindex.py

```diff
@@ -392,13 +393,93 @@ def _popn(self, n):
+
+        if len(self) > max_seq_items:
+            n = int(max_seq_items / 2) + 1
+            top = self[:n]
+            bottom = self[-n:]
+            preprocess = cudf.concat([top, bottom])
+        else:
+            preprocess = self
+
+        preprocess = preprocess.to_pandas(nullable=True)
+        output = preprocess.__repr__()
+        cols_nulls = [
+            self._source_data._data[col].has_nulls
+            for col in self._source_data._data
+        ]
+
+        output_prefix = "MultiIndex("
+        output = output.lstrip(output_prefix)
+        lines = output.split("\n")
+
+        if any(cols_nulls):
+            lines = self.post_process_repr_for_nulls(cols_nulls, lines)
+
+        if len(lines) > 1:
+            if "length=" in lines[-1] and len(self) != len(preprocess):
+                last_line = lines[-1]
+                length_index = last_line.index("length=")
+                last_line = last_line[:length_index] + f"length={len(self)})"
+                lines = lines[:-1]
+                lines.append(last_line)
+
+        data_output = "\n".join(lines)
+        return output_prefix + data_output
+
+    def post_process_repr_for_nulls(self, cols_nulls, lines):
+        row_tuples = list(
+            self._source_data.astype("str")
+            .to_pandas(nullable=True)
+            .itertuples(index=None, name=None)
         )
+        new_lines = []
+        for row_idx, line in enumerate(lines[:-1]):
+            current_row = row_tuples[row_idx]
+            current_row_repr = line.split(", ")
+            for col_idx, value in enumerate(current_row):
+                if value is pd.NA:
+                    if isinstance(
+                        self._source_data._data.columns[col_idx],
+                        (
+                            cudf.core.column.DatetimeColumn,
+                            cudf.core.column.TimeDeltaColumn,
+                        ),
+                    ):
+                        current_row_repr[col_idx] = current_row_repr[
+                            col_idx
+                        ].replace("NaT", "<NA>")
+                    else:
+                        current_row_repr[col_idx] = current_row_repr[
+                            col_idx
+                        ].replace("nan", "<NA>")
+                elif value is np.nan:
+                    # Leave nan's as is.
+                    pass
+                elif cols_nulls[col_idx]:
+                    # Need to pad the values with extra space
+                    # as pandas MultiIndex return 3 character
+                    # string "nan", which are replaced by
+                    # 4 character string "<NA>", hence the
+                    # adjustment is needed for aligned output.
+                    if col_idx == 0:
+                        current_row_repr[col_idx] = current_row_repr[
+                            col_idx
+                        ].replace("(", "( ")
```

Comment:
Can this affect a stringcolumn that has `(` in it in an unexpected way?

Reply:
Changed this replace to happen to only 1st occurrence, so we won't hit this issue. Also added a test-case on these lines.